base/ca/src/main/java/com/netscape/cms/servlet/cert/EnrollServlet.java

```diff
@@ -1280,7 +1285,10 @@ protected void processX509(CMSRequest cmsReq)
             req.setExtData(IRequest.CERT_INFO, certInfoArray);
 
             if (challengePassword != null && !challengePassword.equals("")) {
-                String pwd = hashPassword(challengePassword);
+                MetaInfo metaInfo = (MetaInfo) certInfo.get(CertRecord.ATTR_META_INFO);
+                String challengeString =
```

Comment:
@ckelleyRH Did you actually try to print out what's in challengePassword at this point?  Earlier in CAService.java, the code makes sense because it was doing comparison already, meaning, we expect the challengePassword to already have been hashed and stored in certInfo.  Howeveer, at places (such as here) where the code seems to be preparing for the hashed password to go into the request (an later to be used for comparison, like the code in CAService), so my guess is that you need to take the default hashing alg here to process.  If you haven't already, I suggest you print out challengePassword here to see what you got.  Same goes with other similar places.  I could be wrong of course, but please (temporarily) print out all challengePasswords and pwds involved here so we can make the right decision.

Reply:
@ladycfu - yeah I did try printing the passwords but there was no trace of anything in the log. To check it wasn't the logging being weird I made the `hashPassword()` methods throw `RuntimeException` instead to bring the CI crashing down, but it survived that and all the tests pass. This means that these methods are not being tested in the CI at all, as you feared. The deprecation warning would have disappeared even with `SHA-1` because I moved the `MessageDigest` setup out of the constructors/init methods (which do get called) into `hashPassword()`.

I didn't appreciate before the difference between the preparing and the comparing but i think I understand better now, thanks for the explanation. Now I need to figure out how to trigger this method call in the CI, so that I can see it working, before I try to update my patch. There may well be further questions from me about how to do this.